src/Raven.Server/Documents/PeriodicBackup/GoogleCloudStorage/RavenGoogleCloudStorageClient.cs

```diff
@@ -0,0 +1,158 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Google.Api.Gax;
+using Google.Apis.Auth.OAuth2;
+using Google.Apis.Storage.v1.Data;
+using Google.Apis.Upload;
+using Google.Cloud.Storage.V1;
+using Newtonsoft.Json.Linq;
+using Raven.Client.Documents.Operations.Backups;
+using Object = Google.Apis.Storage.v1.Data.Object;
+
+namespace Raven.Server.Documents.PeriodicBackup.GoogleCloudStorage
+{
+    public class RavenGoogleCloudStorageClient : IDisposable
+    {
+        private readonly StorageClient _client;
+        private readonly string _projectId;
+        protected readonly CancellationToken CancellationToken;
+        private readonly string _bucketName;
+        private readonly Progress _progress;
+
+        private const string ProjectIdPropertyName = "project_id";
+
+        public RavenGoogleCloudStorageClient(string jsonCredential, string bucketName, CancellationToken? cancellationToken = null, Progress progress = null)
+        {
+            try
+            {
+                _client = StorageClient.Create(GoogleCredential.FromJson(jsonCredential));
+            }
+            catch (Exception e)
+            {
+                throw new ArgumentException("Wrong format for account key.", e);
+            }
+
+            if (JObject.Parse(jsonCredential).TryGetValue(ProjectIdPropertyName, StringComparison.InvariantCultureIgnoreCase, out var value))
```

Comment:
Let's split this to two operations. Also use OridinalIgnoreCase comparer

Reply:
Done.